src/main/java/org/kiwiproject/registry/eureka/client/EurekaParser.java

```diff
@@ -18,18 +20,24 @@
 
     @SuppressWarnings("unchecked")
     public static List<EurekaInstance> parseEurekaResponse(Map<String, Object> response) {
-        var applications = ((Map<String, Object>) response.get("applications")).get("application");
+        checkArgumentNotNull(response, "Eureka response map cannot be null");
+
+        var applications = (Map<String, Object>) response.get("applications");
+
+        checkState(nonNull(applications), "Eureka data must contain a key 'applications' that contains a Map<String, Object>");
+
+        var applicationOrApplicationList = applications.get("application");
 
         var eurekaInstances = new ArrayList<EurekaInstance>();
 
-        if (nonNull(applications)) {
-            if (applications instanceof List<?>) {
-                for (Map<String, Object> application : (List<Map<String, Object>>) applications) {
+        if (nonNull(applicationOrApplicationList)) {
+            if (applicationOrApplicationList instanceof List<?>) {
+                for (Map<String, Object> application : (List<Map<String, Object>>) applicationOrApplicationList) {
                     var instances = application.get("instance");
                     eurekaInstances.addAll(parseInstances(instances));
                 }
             } else {
-                eurekaInstances.addAll(parseInstances(((Map<String, Object>) applications).get("instance")));
+                eurekaInstances.addAll(parseInstances(((Map<String, Object>) applicationOrApplicationList).get("instance")));
```

Comment:
Do we want to add a check to make sure it is a Map before the cast here? Or is that overkill?

Reply:
No, not really.  This will hopefully not be arbitrary data.  It is package private and we are only sending in data from Eureka

Reply:
OK.